Author: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Last-Update: 2018-05-25

--- firejail-0.9.54.orig/etc/ardour5.profile
+++ firejail-0.9.54/etc/ardour5.profile
@@ -9,6 +9,10 @@ noblacklist ${HOME}/.config/ardour4
 noblacklist ${HOME}/.config/ardour5
 noblacklist ${HOME}/.lv2
 noblacklist ${HOME}/.vst
+noblacklist ${DOWNLOADS}
+noblacklist ${HOME}/Documents
+noblacklist ${HOME}/Desktop
+noblacklist ${HOME}/Music
 
 include /etc/firejail/disable-common.inc
 include /etc/firejail/disable-devel.inc
@@ -30,7 +34,7 @@ seccomp
 shell none
 
 #private-bin sh,ardour4,ardour5,ardour5-copy-mixer,ardour5-export,ardour5-fix_bbtppq,grep,sed,ldd,nm
-private-dev
+#private-dev
 #private-etc pulse,X11,alternatives,ardour4,ardour5,fonts
 private-tmp
 
