Description: Fix libreoffice profile
Author: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Last-Update: 2018-08-11

--- firejail-0.9.54.orig/etc/libreoffice.profile
+++ firejail-0.9.54/etc/libreoffice.profile
@@ -25,7 +25,7 @@ include /etc/firejail/whitelist-var-comm
 
 # Ubuntu 18.04 uses its own apparmor profile
 # uncomment the next line if you are not on Ubuntu
-#apparmor
+apparmor
 caps.drop all
 machine-id
 netfilter
