Author: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Last-Update: 2018-05-25

--- firejail-0.9.54.orig/etc/uget-gtk.profile
+++ firejail-0.9.54/etc/uget-gtk.profile
@@ -14,6 +14,11 @@ include /etc/firejail/disable-programs.i
 
 mkdir ${HOME}/.config/uGet
 whitelist ${DOWNLOADS}
+whitelist ${HOME}/Desktop
+whitelist ${HOME}/Documents
+whitelist ${HOME}/Pictures
+whitelist ${HOME}/Music  
+whitelist ${HOME}/Video  
 whitelist ${HOME}/.config/uGet
 include /etc/firejail/whitelist-common.inc
 
