--- firejail-0.9.54.orig/etc/android-studio.profile
+++ firejail-0.9.54/etc/android-studio.profile
@@ -15,6 +15,9 @@ noblacklist ${HOME}/.java
 noblacklist ${HOME}/.local/share/JetBrains
 noblacklist ${HOME}/.ssh
 noblacklist ${HOME}/.tooling
+noblacklist ${DOWNLOADS}
+noblacklist ${HOME}/Documents
+noblacklist ${HOME}/Desktop
 
 include /etc/firejail/disable-common.inc
 include /etc/firejail/disable-passwdmgr.inc
