Author: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Last-Update: 2018-05-25

--- firejail-0.9.54.orig/etc/qbittorrent.profile
+++ firejail-0.9.54/etc/qbittorrent.profile
@@ -25,7 +25,12 @@ include /etc/firejail/disable-programs.i
 mkdir ${HOME}/.cache/qBittorrent
 mkdir ${HOME}/.config/qBittorrent
 mkdir ${HOME}/.local/share/data/qBittorrent
-whitelist  ${DOWNLOADS}
+whitelist ${DOWNLOADS}
+whitelist ${HOME}/Desktop
+whitelist ${HOME}/Documents
+whitelist ${HOME}/Pictures
+whitelist ${HOME}/Music
+whitelist ${HOME}/Video
 whitelist ${HOME}/.cache/qBittorrent
 whitelist ${HOME}/.config/qBittorrent
 whitelist ${HOME}/.config/qBittorrentrc
@@ -55,6 +60,6 @@ private-dev
 # private-lib - problems on Arch
 private-tmp
 
-# memory-deny-write-execute - problems on  Arch, see #1690 on GitHub repo
+memory-deny-write-execute
 noexec ${HOME}
 noexec /tmp
