Author: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Last-Update: 2018-05-25

--- firejail-0.9.54.orig/etc/thunderbird.profile
+++ firejail-0.9.54/etc/thunderbird.profile
@@ -21,6 +21,9 @@ whitelist ${HOME}/.cache/thunderbird
 whitelist ${HOME}/.gnupg
 # whitelist ${HOME}/.icedove
 whitelist ${HOME}/.thunderbird
+whitelist ${DOWNLOADS}
+whitelist ${HOME}/Desktop
+whitelist ${HOME}/Documents
 
 # We need the real /tmp for data exchange when xdg-open handles email attachments on KDE
 ignore private-tmp
