Description: Fix missing python support in playonlinux
Author: CoffeeFou <coffeefou@protonmail.com>
Last-Update: 2018-11-17

--- firejail-0.9.56.orig/etc/playonlinux.profile
+++ firejail-0.9.56/etc/playonlinux.profile
@@ -24,8 +24,8 @@ noblacklist /usr/share/perl*
 
 include /etc/firejail/disable-common.inc
 # playonlinux uses perl
-include /etc/firejail/disable-devel.inc
-include /etc/firejail/disable-interpreters.inc
+# include /etc/firejail/disable-devel.inc
+# include /etc/firejail/disable-interpreters.inc
 include /etc/firejail/disable-programs.inc
 
 caps.drop all
