Description: Fix discord profile
Author: Coffeefou
Reviewed-By: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Last-Update: 2018-07-11

Index: firejail-0.9.54/etc/discord.profile
===================================================================
--- firejail-0.9.54.orig/etc/discord.profile
+++ firejail-0.9.54/etc/discord.profile
@@ -24,9 +24,9 @@ novideo
 protocol unix,inet,inet6,netlink
 seccomp
 
-private-bin discord,sh,xdg-mime,tr,sed,echo,head,cut,xdg-open,grep,egrep
+#private-bin discord,sh,xdg-mime,tr,sed,echo,head,cut,xdg-open,grep,egrep
 private-dev
-private-etc fonts,machine-id
+#private-etc fonts,machine-id
 private-tmp
 
 noexec ${HOME}
