Description: Fix atril profile
Author: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Last-Update: 2018-05-25

--- firejail-0.9.54.orig/etc/atril.profile
+++ firejail-0.9.54/etc/atril.profile
@@ -37,13 +37,12 @@ tracelog
 
 private-bin atril, atril-previewer, atril-thumbnailer
 private-dev
-private-etc fonts,ld.so.cache
+private-etc fonts,ld.so.cache,alternatives
 # atril uses webkit gtk to display epub files
 # waiting for globbing support in private-lib; for now hardcoding it to webkit2gtk-4.0
-#private-lib webkit2gtk-4.0 - problems on Arch with the new version of WebKit
+private-lib webkit2gtk-4.0
 private-tmp
 
-# webkit gtk killed by memory-deny-write-execute
-#memory-deny-write-execute
+memory-deny-write-execute
 noexec ${HOME}
 noexec /tmp
