Author: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Last-Update: 2018-05-25

Index: firejail-0.9.54/etc/virtualbox.profile
===================================================================
--- firejail-0.9.54.orig/etc/virtualbox.profile
+++ firejail-0.9.54/etc/virtualbox.profile
@@ -21,6 +21,8 @@ mkdir ${HOME}/VirtualBox VMs
 whitelist ${HOME}/.config/VirtualBox
 whitelist ${HOME}/VirtualBox VMs
 whitelist ${DOWNLOADS}
+whitelist ${HOME}/Desktop
+whitelist ${HOME}/Documents
 include /etc/firejail/whitelist-common.inc
 include /etc/firejail/whitelist-var-common.inc
 
