Description: Fix code profile
Author: Coffeefou
Reviewed-By: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Last-Update: 2018-07-11

--- firejail-0.9.54.orig/etc/code.profile
+++ firejail-0.9.54/etc/code.profile
@@ -13,7 +13,8 @@ include /etc/firejail/disable-passwdmgr.
 include /etc/firejail/disable-programs.inc
 
 caps.drop all
-net none
+#uncomment these lines to block internet access
+#net none
 netfilter
 nodvd
 nogroups
