Description: Libreoffice needs clang at first boot.
Author: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Last-Update: 2017-12-01

--- firejail-0.9.50.orig/etc/libreoffice.profile
+++ firejail-0.9.50/etc/libreoffice.profile
@@ -10,7 +10,10 @@ noblacklist /usr/local/sbin
 noblacklist ~/.config/libreoffice
 
 include /etc/firejail/disable-common.inc
-include /etc/firejail/disable-devel.inc
+
+# libreoffice needs it at first boot
+#include /etc/firejail/disable-devel.inc
+
 include /etc/firejail/disable-passwdmgr.inc
 include /etc/firejail/disable-programs.inc
 
