Description: Don't blacklist local cache
Author: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Last-Update: 2017-12-10

--- firejail-0.9.50.orig/etc/libreoffice.profile
+++ firejail-0.9.50/etc/libreoffice.profile
@@ -8,7 +8,7 @@ include /etc/firejail/globals.local
 noblacklist ${HOME}/.java
 noblacklist /usr/local/sbin
 noblacklist ~/.config/libreoffice
-
+noblacklist ~/.cache
 include /etc/firejail/disable-common.inc
 
 # libreoffice needs it at first boot
